ogr/mock_core.py

```diff
@@ -0,0 +1,124 @@
+from typing import Callable, Any, Optional
+import functools
+import logging
+import datetime
+from ogr.abstract import PullRequest, PRComment, PRStatus, GitProject
+from ogr.constant import DEFAULT_RO_PREFIX_STRING
+
+
+logger = logging.getLogger(__name__)
+
+
+def readonly(
+    *,
+    return_value: Optional[Any] = None,
+    return_function: Optional[Callable] = None,
+    log_message: str = "",
+) -> Any:
+    """
+    Decorator to log  function and ovewrite return value of object methods
+    Ignore function name as first parameter and ignore every other parameters
+
+    :param return_value: returned Any value if given, return_function has higher prio if set
+    :param return_function: return function and give there parameters also
+           original caller object return_function(self, *args, **kwargs)
+    :param log_message: str string to put to logger output
+    :return: Any type what is expected that function or return value returns
+    """
+
+    def decorator_readonly(func):
+        @functools.wraps(func)
+        def readonly_func(self, *args, **kwargs):
+
+            output_str = DEFAULT_RO_PREFIX_STRING
+            if not self.read_only:
+                return func(self, *args, **kwargs)
+            else:
+                if log_message:
+                    output_str += " " + log_message
+                args_str = str(args)[1:-1]
+                kwargs_str = ""
+                for k in kwargs:
+                    kwargs_str += f", {k}={kwargs[k]!r}"
```

Comment:
This will make `kwargs_str` start with ", ". What you want is:

```python
kwargs_str = ", ".join(f"{k}={v!r}" for k,v in kwargs.items())
```

Reply:
you are right, but not sure about readability. but that's true that also current code is not very well readable :-)